### DISCUSSION TITLE
difx:Driver child element

### DISCUSSION THREAD

**Discussion by computerquip-work** [wixtoolset/issues#7878] [Questions]
I've been trying to get my MSI installer to control the installation of some drivers. I'm having a heck of a time with difx:Driver unfortunately. While I'm aware it's deprecated, as far as I understand, there's no alternative currently. If that is not the case, **please** let me know.

```
      <Component>
        <difx:Driver ForceInstall="no"/>
        <File Name="xyz-device.inf" />
      </Component>
```
I have something akin to this. I have three of these. Two of them are in a separate feature. They compile and work perfectly fine. The third one fails with `error WIX0207: The Component element contains an unexpected child element 'Driver'. The 'Driver'  element may only occur 1 time(s) under the Component element`. This error makes no sense unfortunately.

While I'd like to post the whole schema, I can't currently so I'm hoping this is a known issue and there's a work around.

EDIT:
With the child element in place, I get exactly three warnings of `warning WIX1130: The Driver element has been deprecated.`. If I comment it out, I get two. This seems like a bug behind the scenes. :/

**Comment by computerquip-work** [comment]
So, it turns out there's some component collision going on which is a bit concerning. In the above example, there is no other filename named "xyz-device.inf" and even if it's marked as an explicit KeyPath, it still doesn't work. If I mark the component with an explicit id, it does work, even if I mark it with what I thought the identifier would have been i.e. `xyz_device.inf`.

This sort of concerns me since it goes against my understanding of how the component identifiers are generated implicitly. Any clarification on that would be appreciated.

**Reply by wmanning** [reply]
I use difx:Driver with no issues. I have one driver per MSI though.

**Comment by pas059** [comment]
Hello,
on my side, to install a driver with a wix installer, i use 'pnputil.exe' and a [quiet execution custom action](https://wixtoolset.org/docs/tools/wixext/quietexec/).
